backend/de.metas.material/dispo-service/src/main/java/de/metas/material/dispo/service/candidatechange/StockCandidateService.java

```diff
@@ -168,69 +174,63 @@ public CandidateSaveResult updateQtyAndDate(@NonNull final Candidate candidateTo
 	 * Selects all stock candidates which have the same product and locator but a later timestamp than the one from the given {@code materialDescriptor}.
 	 * Iterate them and add the given {@code delta} to their quantity.
 	 */
-	public void applyDeltaToMatchingLaterStockCandidates(@NonNull final CandidateSaveResult stockWithDelta)
+	public void applyDeltaToMatchingLaterStockCandidates(@NonNull final CandidateSaveResult saveResult)
 	{
-		final Candidate deltaCandidate = stockWithDelta.getCandidate();
-		if (deltaCandidate.isSimulated())
+		final Candidate initialStockCandidate = saveResult.getCandidate();
+		if (initialStockCandidate.isSimulated())
 		{
 			return;
 		}
 
-		final CandidatesQuery query = createStockQueryBetweenDates(stockWithDelta);
-
-		final BigDecimal deltaUntilRangeEnd;
-		final BigDecimal deltaAfterRangeEnd;
-		if (stockWithDelta.isDateMoved())
-		{
-			if (stockWithDelta.isDateMovedForwards())
-			{
-				Check.assumeNotNull(stockWithDelta.getPreviousQty(), "PreviousQty cannot be null on update case!");
-
-				deltaUntilRangeEnd = stockWithDelta.getPreviousQty().negate();
-				deltaAfterRangeEnd = stockWithDelta.getQtyDelta();
-			}
-			else
-			{
-				deltaUntilRangeEnd = deltaCandidate.getQuantity();
-				deltaAfterRangeEnd = stockWithDelta.getQtyDelta();
-			}
-		}
-		else
+		final CandidatesQuery query = createStockQueryAfterDate(saveResult);
+		final List<Candidate> stockCandidatesToUpdate = candidateRepositoryRetrieval.retrieveOrderedByDateAndSeqNo(query);
+		if (stockCandidatesToUpdate.isEmpty())
 		{
-			deltaUntilRangeEnd = stockWithDelta.getQtyDelta();
-			deltaAfterRangeEnd = null;
+			return;
 		}
 
-		final List<Candidate> candidatesToUpdateWithinRange = candidateRepositoryRetrieval.retrieveOrderedByDateAndSeqNo(query);
-		for (final Candidate candidate : candidatesToUpdateWithinRange)
+		final ImmutableMap<CandidateId, Candidate> stockIdToMainCandidateMap = computeStockIdToMainCandidateMap(stockCandidatesToUpdate);
+
+		Candidate previousStockCandidate = getPreviousStockOrNullForCandidate(stockCandidatesToUpdate.get(0));
+		final MaterialDispoGroupId groupId = initialStockCandidate.getGroupId();
+		for (final Candidate stockCandidate : stockCandidatesToUpdate)
 		{
-			//TODO refactor in next iteration to use MD_Candidate_QtyDetails
-			final BigDecimal newQty = candidate.getQuantity().add(deltaUntilRangeEnd);
-			candidateRepositoryWriteService.updateCandidateById(candidate
+			final Candidate mainCandidate = stockIdToMainCandidateMap.get(stockCandidate.getId());
+			final BigDecimal previousStockQty = previousStockCandidate == null ? BigDecimal.ZERO : previousStockCandidate.getQuantity();
+			final BigDecimal newQty = previousStockQty.add(mainCandidate.getStockImpactPlannedQuantity());
+			candidateRepositoryWriteService.updateCandidateById(stockCandidate
```

Comment:
IMHO, u shall update the details and then from details calculate the new qty

Reply:
Changed to use qty resulted from details update.